static/styles/zulip.css

```diff
@@ -73,6 +73,21 @@ a {
     }
 }
 
+input[type="radio"] {
+    width: auto;
+    cursor: pointer;
+
+    &:focus {
+        outline: 1px dotted hsl(0, 0%, 20%);
+        outline: 5px auto -webkit-focus-ring-color;
+        outline-offset: -2px;
+    }
+
+    &:disabled {
+        cursor: not-allowed;
+    }
+}
```

Comment:
So a big part of the goal here is not to have `zulip.css` CSS for all radio inputs -- we're trying to replace that CSS style with reusable components, like "settings_checkbox", that have a specific style. 

I think the right model is to have a `input[type="radio"].settings-radio-input` class, and use that in all the settings contexts; and then be OK with a bit of duplication with the one non-settings-styled use case (the enter-sends widget). 

We don't have to go all the way there in this PR, but the direction should be that rather than having CSS for a list of specific settings ratio elements like this:
```
.emojiset_choices,
.user_list_style_values
```

We'd just have CSS for a `settings-radio-input` object, and an HTML partial template that can be included when one wants such a component in the app. Or maybe we end up with two types -- a simple radio input and a `display_settings_radio_input` class that's for the ones where we show a visual preview. 

In particular, we want to eliminate all CSS that applies to all radio inputs, since it means that any radio input that happens to want different styling needs to override the others.

Does that make sense?

Reply:
I think probably this model suggests a different commit structure -- a commit could introduce a new common class and use it in one place, and later commits could use it in more places; or maybe even better given there's only 2 settings radio buttons with the preview, it'd probably be fine to just do a commit that makes the class and changes both of them to use it. And then another for variants.

Reply:
For the CSS in `zulip.css` my thinking is that atleast the `cursor` property would be constant across the app (and width will probably be removed once we remove bootstrap for all type of inputs), so would be better to not duplicate them. But yeah, the model of reusable components you described above sounds good.

There are also differences between radio inputs used in stream settings and org settings since the stream settings have multil-ine label, so would require some CSS changes to have a common class (and thus CSS) for both types of settings or separate class. And also the margin values are different in different cases depending upon the label. Will probably need some tweaking in rules too if we want to avoid having multiple components. Will look into it.

Reply:
I think two separate classes is fine for the two styles -- there's no need to try to fit the same class on two different components. Is the `cursor` property no the default for browsers (you can check looking at the `user-agent` stylesheets)? It's possible it's just an unnecessary line.

Reply:
The browser value is `cursor: default` and we want `cursor: pointer`.